app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -147,6 +150,25 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
         browserInitialized = initializeUI(view) != null
     }
 
+    private fun animateBrowserEngine(browserEngine: View) {
```

Comment:
I think this might be easier to read if `animateBrowserEngine` and `captureEngineViewAndDrawStatically` were moved to a new class, and they both seem relatively easy to move.  It would let us avoid bloating the fragment, group the constants a little more logically, etc.

Reply:
I agree these should be moved to a new class out of the fragment. 

Also in theory these shouldn't be needed in `BaseBrowserFragment` only `BrowserFragment` because we will only animate in normal browsing not custom tabs, right? Not sure if the timing works out the same, but if nothing else we should remove any delays or things that would impact custom tab or action view intent performance and are unneeded 🤷‍♀️

Reply:
I have refactored this logic all into `BrowserAnimator.kt`. 

@ekager I actually do need to have the animation in BaseBrowserFragment because I pass the "captureEngineView" logic to the browserToolbar so it can play it as needed (to prevent "flickering" of GV). That being said, I am **only delaying** if the newly added `shouldAnimate` flag is set (which defaults to false) and is only set on Home -> Browser transition. This will avoid there being any delays in custom tabs or on resuming the fragment 😄